.github/workflows/sonar-cloud.yml

```diff
@@ -0,0 +1,63 @@
+name: Check format and coverage, then upload to SonarCloud
+
+on:
+  push:
+    branches:
+      - main
+      - dev
+    paths:
+      - bin/**
+      - src/**
+      - tests/**
+      - setup.py
+      - .github/workflows/build.yml
+  pull_request:
+    branches:
+      - main
+      - dev
+    paths:
+      - bin/**
+      - src/**
+      - tests/**
+      - setup.py
+      - .github/workflows/build.yml
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
+jobs:
+
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.10'
+      - name: Python info
+        shell: bash -l {0}
+        run: |
+          which python3
+          python3 --version
+      - name: Upgrade pip and setuptools
+        run: |
+          python3 -m pip install --upgrade pip setuptools wheel
+      - name: Install nplinker and its dependencies
+        run: |
+          python3 -m pip install .[dev]
+          install-nplinker-deps --run-on-github
+      - name: Check style against standards using ruff
+        run: ruff check .
+      - name: Run unit tests with coverage
+        run: pytest --cov --cov-report term --cov-report xml --junitxml=xunit-result.xml tests/unit
+      - name: Correct coverage paths
```

Comment:
Why is this correction needed?

Reply:
The `coverage.xml` has absolute paths to code files. This correction is to change absolute paths to relative paths by removing the `$PWD`.